### ISSUE TITLE
Feature request: user-defined number type

### ISSUE THREAD

**Issue by data-man** [Vexu/bog#2]
E.g.:
```zig
pub fn Vm(comptime intType: type, comptime floatType: type) {
 ...
}

var vm = bog.Vm(u128, f128).init(allocator, .{ .import_files = true });
```

**Comment by Vexu** [comment]
I don't think this is useful enough to justify making `Vm` generic, but it could be provided as a build option.

**Comment by data-man** [comment]
> it could be provided as a build option.

:+1: 

Are you interested in extending the standard library?
E.g. `std.*` or `std.math.*`.

Or should I open a separate issue for that?

**Comment by Vexu** [comment]
> Are you interested in extending the standard library?

Sure, I haven't got around to do much with the std since other parts like the gc are still quite broken.

> should I open a separate issue for that?

Not necessary, but feel free to do so.
